beacon/validator/block_builder.go

```diff
@@ -95,9 +93,6 @@ func (s *Service) BuildBlockAndSidecars(
 	if err != nil {
 		return nil, nil, err
 	}
-	if envelope == nil {
```

Comment:
this check is not necessary since all code path leading here do intercept the condition `envelope == nil` and return an error